invenio_drafts_resources/services/records/components/files.py

```diff
@@ -47,25 +59,35 @@ def update_draft(self, identity, data=None, record=None, errors=None):
         draft = record
         enabled = data.get("files", {}).get("enabled", True)
         default_preview = data.get("files", {}).get("default_preview")
-
-        try:
-            self.files_component.assign_files_enabled(draft, enabled)
-        except ValidationError as e:
-            errors.append({"field": "files.enabled", "messages": e.messages})
-            return  # exit early
-
-        if draft.files.enabled and not draft.files.items():
+        if self.service.check_permission(identity, "manage_files"):
+            try:
+                self.files_component.assign_files_enabled(draft, enabled)
+            except ValidationError as e:
+                errors.append({"field": "files.enabled", "messages": e.messages})
+                return  # exit early
+
+            if draft.files.enabled and not draft.files.items():
+                errors.append(
+                    {
+                        "field": "files.enabled",
```

Comment:
Very minor: should the field may be only `files` in this case, and not `files.enabled`?